File: app/src/main/java/com/jintin/app/BaseActivity.java
package com.jintin.app;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.RecyclerView;

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_recycler);
        RecyclerView recyclerView = findViewById(R.id.recyclerView);
        setupRecyclerView(recyclerView);
        recyclerView.setAdapter(getAdapter());
    }

    protected abstract void setupRecyclerView(RecyclerView recyclerView);

    protected abstract RecyclerView.Adapter getAdapter();
}

File: app/src/main/java/com/jintin/app/BaseActivity.kt
package com.jintin.app

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.RecyclerView

abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_recycler)
        val recyclerView = findViewById<RecyclerView>(R.id.recyclerView)
        setupRecyclerView(recyclerView)
        recyclerView.adapter = getAdapter()
    }

    protected abstract fun setupRecyclerView(recyclerView: RecyclerView)

    protected abstract fun getAdapter(): RecyclerView.Adapter<*>
}
